keylime/cloud_verifier_tornado.py

```diff
@@ -293,77 +294,71 @@ def put(self):
 
 
     def invoke_get_quote(self, agent, need_pubkey):
+        if agent is None:
+            raise Exception("agent deleted while being processed")
         params = cloud_verifier_common.prepare_get_quote(agent)
         agent['operational_state'] = cloud_verifier_common.CloudAgent_Operational_State.GET_QUOTE
-        client = tornado.httpclient.AsyncHTTPClient()
 
         partial_req = "1"
         if need_pubkey:
             partial_req = "0"
 
-        url = "http://%s:%d/quotes/integrity?nonce=%s&mask=%s&vmask=%s&partial=%s"%(agent['ip'],agent['port'],params["nonce"],params["mask"],params['vmask'],partial_req)
-        # the following line adds the agent and params arguments to the callback as a convenience
-        cb = functools.partial(self.on_get_quote_response, agent, url)
-        client.fetch(url, callback=cb)
-
-    def on_get_quote_response(self, agent, url, response):
-        if agent is None:
-            raise Exception("agent deleted while being processed")
-        if response.error:
-            # this is a connection error, retry get quote
-            if isinstance(response.error, IOError) or (isinstance(response.error, tornado.web.HTTPError) and response.error.code == 599):
-                self.process_agent(agent, cloud_verifier_common.CloudAgent_Operational_State.GET_QUOTE_RETRY)
+        while True:
+            params = f'/quotes/integrity?nonce={params["nonce"]}&mask={params["mask"]}&vmask={params["vmask"]}&partial={partial_req}'
```

Comment:
It looks like the existing `params` variable (a dictionary) is being overwritten here (with a string).  Recommend renaming to `path` (and updating below to `params=path`).

Reply:
Ah yes, good catch!

Reply:
Does this solve the issue you described above, or is that unrelated?